### PR TITLE
Add an example of how to parallel builds on multiple nodes

### PR DESCRIPTION
**Pull request by omehegan**

### CODE REVIEW

File: pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy [hunk 1]
```diff
@@ -0,0 +1,12 @@
+def labels = ['precise', 'trusty'] // labels for Jenkins node types we will build on
+def builders = [:]
+for (x in labels) {
```
Comment (rtyler) [inline pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy]: Is there a reason to not to define this as `for (label in labels) {` and skip the `def label = x` below?

File: pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    def label = x
+    builders[label] = {
```
Comment (rtyler) [inline pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy]: I suggest adding a comment here that says something to the effect of "building a map to pass into the `parallel` step so we can fire off all the builds at once`

File: pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy (hunk 1 above)
Comment (omehegan) [inline pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy]: I did that and it didn't work.
12:37 <+jglick> autojack: the subject of endless duplicates on stackoverflow.com/questions/tagged/jenkins-workflow. Actually a Groovy question, not Pipeline specific. Short answer: bind `label` to a fresh variable outside the closure.

File: pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy (hunk 2 above)
Comment (omehegan) [inline pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy]: Done.

File: pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy (hunk 1 above)
Comment (rtyler) [inline pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy]: So it's a Groovy scoping issue inside the nested closure? Jesse's comment doesn't actually explain anything to me :frowning: 

Anyways, I suggest offering a little comment explaining why the variable needs to be bound, lest somebody like me remove it and be confused when things explode

Comment (omehegan) [inline pipeline-examples/parallel-multiple-nodes/ParallelMultipleNodes.groovy]: When I tried it the other way, I got two parallel builds on the same label. I'll add a comment.

**Pull request merged**